### PR TITLE
Issue 530 JMAP endpoint should support being called with accountIds of delegated accounts

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/530

### CODE REVIEW

**Review by chibenwa — COMMENTED** [review]
Forward/get is likely OK IMO...

File: tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLongLivedTokenGetTest.java [hunk 1]
```diff
@@ -11,11 +12,11 @@
 import com.linagora.tmail.james.app.DistributedServer;
 import com.linagora.tmail.james.app.DockerOpenSearchExtension;
 import com.linagora.tmail.james.app.RabbitMQExtension;
-import com.linagora.tmail.james.common.LinagoraLongLivedTokenSetMethodContract;
+import com.linagora.tmail.james.common.LongLivedTokenGetContract;
 import com.linagora.tmail.james.common.module.JmapGuiceKeystoreManagerModule;
 import com.linagora.tmail.module.LinagoraTestJMAPServerModule;
 
-public class DistributedLongLivedTokenGetTest implements LinagoraLongLivedTokenSetMethodContract {
+public class DistributedLongLivedTokenGetTest implements LongLivedTokenGetContract {
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLongLivedTokenGetTest.java:19]: good catch

**Review by Arsnael — APPROVED** [review]
Answer @chibenwa comment please, but otherwise I'm ok :)

**Comment by vttranlina** [comment]
> Answer @chibenwa comment please, but otherwise I'm ok :)

I thought he said it's ok
I miss somewhere?

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/ForwardGetMethod.scala [hunk 2]
```diff
@@ -84,6 +84,7 @@ class ForwardGetMethod @Inject()(recipientRewriteTable: RecipientRewriteTable,
   override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_CORE, LINAGORA_FORWARD)
 
   override def doProcess(capabilities: Set[CapabilityIdentifier], invocation: InvocationWithContext, mailboxSession: MailboxSession, request: ForwardGetRequest): SMono[InvocationWithContext] = {
+    DelegatedAccountPrecondition.acceptOnlyOwnerRequest(mailboxSession, request.accountId)
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/ForwardGetMethod.scala:87]: I mean we likely could allow Bob to see Alice forward, not manage it.

```suggestion
```

**Comment by Arsnael** [comment]
I think he meant Forward/Get is likely OK to be called by a delegated account... It wasn't clear but implicit. Otherwise he would have approved the PR on the moment with his comment I think

@chibenwa can you confirm please? Thanks :)

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**